src/main/java/org/skriptlang/skript/bukkit/recipes/elements/ExprRecipeIngredients.java

```diff
@@ -0,0 +1,473 @@
+			"This can only be used when creating a Shaped Recipe."),
+		SECONDROW("recipe ingredients of [the] (2nd|second) row", "recipe ingredients of the first row", ShapedRecipeEvent.class,
+			"This can only be used when creating a Shaped Recipe."),
+		THIRDROW("recipe ingredients of [the] (3rd|third) row", "recipe ingredients of the first row", ShapedRecipeEvent.class,
+			"This can only be used when creating a Shaped Recipe."),
+		INPUT("recipe (input|source) [item]", "recipe input item", new Class[]{CookingRecipeEvent.class, StonecuttingRecipeEvent.class, TransmuteRecipeEvent.class},
+			"This can only be used when creating a Cooking, Blasting, Furnace, Campfire, Smoking, Stonecutting, or Transmute Recipe."),
+		BASE("[recipe] base item[s]", "recipe base items", SmithingRecipeEvent.class,
+			"This can only be used when creating a Smithing, Smithing Transform, or Smithing Trim Recipe."),
+		TEMPLATE("[recipe] template item[s]", "recipe template items", new Class[]{SmithingTransformRecipeEvent.class, SmithingTrimRecipeEvent.class},
+			"This can only be used when creating a Smithing Transform or Smithing Trim Recipe."),
+		ADDITION("[recipe] addition[al] item[s]", "recipe additional items", SmithingRecipeEvent.class,
+			"This can only be used when creating a Smithing, Smithing Transform or Smithing Trim Recipe."),
+		TRANSMUTE("[recipe] transmute item[s]", "recipe transmute items", TransmuteRecipeEvent.class,
+			"This can only be used creating a Transmute Recipe.");
+
+
+		private String pattern, pattern2, toString, error;
+		private Class<? extends Event>[] eventClasses;
+
+		RecipePattern(String pattern, String toString, Class<? extends Event> eventClass, String error) {
+			//noinspection unchecked
+			this(pattern, toString, new Class[]{eventClass}, error);
+		}
+
+		RecipePattern(String pattern, String toString, Class<? extends Event>[] eventClasses, String error) {
+			this.pattern = "[the] " + pattern + " [of %recipes%]";
+			this.pattern2 = "[the] %recipes%'[s] " + pattern;
+			this.toString = toString;
+			this.eventClasses = eventClasses;
+			this.error = error;
+		}
+
+	}
+
+	private static final RecipePattern[] recipePatterns = RecipePattern.values();
+
+	static {
+		String[] patterns = new String[recipePatterns.length * 2];
+		for (RecipePattern pattern : recipePatterns) {
+			patterns[(2 * pattern.ordinal())] = pattern.pattern;
+			patterns[(2 * pattern.ordinal()) + 1] = pattern.pattern2;
+		}
+		Skript.registerExpression(ExprRecipeIngredients.class, ItemStack.class, ExpressionType.PROPERTY, patterns);
+	}
+
+	private boolean isEvent = false;
+	private RecipePattern selectedPattern;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		selectedPattern = recipePatterns[(int) Math.floor((double) matchedPattern / 2)];
```

Comment:
```suggestion
		selectedPattern = recipePatterns[matchedPattern / 2];
```
integer division should truncate for you